### PR TITLE
Several SPM related fixes

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

Part of #8123.

### CODE REVIEW

File: plugins/package-managers/spm/src/main/kotlin/Spm.kt [hunk 1]
```diff
@@ -141,13 +141,13 @@ class Spm(
...
         val vcsInfo = VersionControlSystem.forDirectory(definitionFile.parentFile)?.getInfo().orEmpty()
-        val (author, project) = parseAuthorAndProjectFromRepo(repositoryURL = vcsInfo.url)
+        val (author, _) = parseAuthorAndProjectFromRepo(repositoryURL = vcsInfo.url)
 
         val projectIdentifier = Identifier(
             type = managerName,
             version = vcsInfo.revision,
             namespace = author.orEmpty(),
-            name = project ?: definitionFile.parentFile.relativeTo(analysisRoot).invariantSeparatorsPath
+            name = definitionFile.parentFile.relativeTo(analysisRoot).invariantSeparatorsPath
```
Comment (sschuberth) [inline plugins/package-managers/spm/src/main/kotlin/Spm.kt:150]: IMO we should strive to use the `getFallbackProjectName()` helper function like @mnonnenmacher recently did as part of https://github.com/oss-review-toolkit/ort/pull/8117.

File: plugins/package-managers/spm/src/main/kotlin/SpmModel.kt [hunk 2]
```diff
@@ -85,11 +85,11 @@ data class LibraryDependency(
 
     override val id: Identifier
         get() {
-            val (author, project) = parseAuthorAndProjectFromRepo(repositoryUrl)
+            val (author, _) = parseAuthorAndProjectFromRepo(repositoryUrl)
```
Comment (sschuberth) [inline plugins/package-managers/spm/src/main/kotlin/SpmModel.kt:86]: IMO it's hard to understand why this is correct without looking up the implementation of `parseAuthorAndProjectFromRepo()`. What would have helped me during the review is a sentence like this in the commit message:

"The `parseAuthorAndProjectFromRepo()` function tries to parse the project name from the repository URL. However, this requires the URL to point to a known VCS host, and generally the dedicated `name` field from the SPM data model should be used instead."

**Review by sschuberth — CHANGES_REQUESTED** [review]
I would have expected more clean-up commits regarding `parseAuthorAndProjectFromRepo()` in this PR, as all callers now only seem to use the `author`. On top of that, the `author` is also not used as an "author" anymore, but as an organization /  namespace.

So how about changing that function to just `parseNamespaceFromRepo()`?

**Comment by fviernau** [comment]
> I would have expected more clean-up commits regarding `parseAuthorAndProjectFromRepo()` in this PR, as all callers now only seem to use the `author`. On top of that, the `author` is also not used as an "author" anymore, but as an organization / namespace.
> 
> So how about changing that function to just `parseNamespaceFromRepo()`?

I prefer to do so in a following PR, or rather remove the function entirely instead. (I believe it's likely this will become possible with upcoming changes)

**Review by sschuberth — APPROVED** [review]

**Pull request merged**